sections/header.liquid

```diff
@@ -89,6 +89,7 @@
   }
 
   .section-header {
+    position: sticky;
```

Comment:
Sorry Eugene I haven't had to time to fully test/think on this just yet, but wanted to make sure I remembered to mention that we should definitely add a comment to indicate this is for fixing a safari z-index issue. Maybe include the PR number #2147 in the comment as well